packages/core/src/compiler/Style.ts

```diff
@@ -608,6 +676,126 @@ const couldMatchRels = (
   return true;
 };
 
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param sebind
+ * exist as valid keys in @param subst.
+ */
+const sebindExistsInSubst = (sebind: SEBind, subst: Subst): boolean => {
+  if (sebind.contents.tag !== "StyVar") {
+    throw new Error("expected sty var");
+  }
+  const validKeywords = Object.keys(subst);
+  return validKeywords.includes(sebind.contents.contents.value);
+};
+
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param expr
+ * exist as valid keys in @param subst.
+ */
+const selExprExistsInSubst = (expr: SelExpr, subst: Subst): boolean => {
+  if (expr.tag === "SEBind") {
+    return sebindExistsInSubst(expr, subst);
+  } else if (expr.tag === ("SEFunc" || "SEValCons" || "SEFuncOrValCons")) {
+    return expr.args.every((arg) => selExprExistsInSubst(arg, subst));
+  } else throw new Error("unknown tag");
+};
+
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param arg
+ * exist as valid keys in @param subst.
+ */
+const predArgExistsInSubst = (arg: PredArg, subst: Subst): boolean => {
+  if (arg.tag === "RelPred") {
+    return isValidRelSubst(subst, arg);
+  } else if (arg.tag === "SEBind") {
+    return sebindExistsInSubst(arg, subst);
+  } else throw new Error("unknown tag");
+};
+
+/**
+ * Determines if all the variables in @param rel exist as valid
+ * keys in @param subst.
+ *
+ * Ex, IsSubset(x,y) must have 'x' and 'y' exist as keys in @param subst
+ * for the relation to be valid.
+ */
+const isValidRelSubst = (subst: Subst, rel: RelationPattern): boolean => {
+  if (rel.tag === "RelPred") {
+    return rel.args.every((arg) => predArgExistsInSubst(arg, subst));
+  } else if (rel.tag === "RelBind") {
+    return false; // these don't have aliases
+  } else {
+    throw new Error("unknown tag");
+  }
+};
+
+/**
+ * Helper fxn for @function getRelPredAliasInstanceName
+ *
+ * Returns the substitution for a bindingform
+ */
+const getBindingFormAliasInstanceName = (
+  bf: BindingForm,
+  subst: Subst
+): string => {
+  if (bf.tag === "SubVar") {
+    return bf.contents.value;
+  } else if (bf.tag === "StyVar") {
+    return subst[bf.contents.value];
+  } else throw new Error("unknown tag");
+};
+
+/**
+ * Returns the substitution for a predicate alias
+ */
+// IsSubset(B,A) --> `IsSubset_B_A`
+// IsSubset(Union(B,C),A) --> `IsSubset_Union_B_C_A`
+// TODO: this can be refactored for a more descriptive name for nested cases
+// TODO: adding parentheses into the strings messes with GPIs sometimes?
```

Comment:
> // TODO: this can be refactored for a more descriptive name for nested cases
Right now I'm creating predicate names by replacing parentheses and commas with underscores, so if the user declares `IsSubset(Union(B,C),A)` in substance and assigns some objects to this predicate in style, then the key for the predicate instance in the translation's `trMap` will be `IsSubset_Union_B_C_A`. I think this sort of mapping is okay for now, but I'm not sure that it's bijective between the predicate instance (`IsSubset(Union(B,C),A)`) and the translation key (`IsSubset_Union_B_C_A`), which is why a more descriptive mapping that differentiates between parens and commas (esp for nested cases) might be good. 

> // TODO: adding parentheses into the strings messes with GPIs sometimes?
Ideally I would directly use the substance statement `IsSubset(Union(B,C),A)` as the key in the translation, but using parentheses in the key seems to mess with the shape GPIs sometimes (it might just be this one example)

`npx roger watch tree.sub tree.sty setTheory.dsl` in `__tests__/predicate_aliasing_examples`

Using all underscores in the translation keys:
<img width="1680" alt="Screen Shot 2021-08-03 at 4 06 55 PM" src="https://user-images.githubusercontent.com/59301744/128079024-e94bfe61-3a51-4102-8e88-686635f8ea61.png">

Using parentheses in the translation keys: (arrows missing their heads?)
<img width="1680" alt="Screen Shot 2021-08-03 at 4 07 32 PM" src="https://user-images.githubusercontent.com/59301744/128079089-f7efb57e-43bc-4984-afef-24cb741a973e.png">

Reply:
> Ideally I would directly use the substance statement `IsSubset(Union(B,C),A)` as the key in the translation, but using parentheses in the key seems to mess with the shape GPIs sometimes (it might just be this one example)

Yeah parens in string keys might not be the greatest idea... Not sure exactly how it'd cause problems tho